src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -2230,14 +2208,30 @@ int os::open(const char *path, int oflag, int mode) {
   // 4843136: (process) pipe file descriptor from Runtime.exec not being closed
   // 6339493: (process) Runtime.exec does not close all file descriptors on Solaris 9
   //
-#ifdef FD_CLOEXEC
+
+  // O_CLOEXEC is preferable to using FD_CLOEXEC on an open file descriptor
+  // because it saves a system call and removes a small window where the flag
+  // is unset.
+  int fd = ::open(path, oflag|O_CLOEXEC, mode);
```

Comment:
Style nit: should there be spaces between |?

Reply:
Done.